### PR TITLE
Alien weed fix and further fixes to the broken smoothing system

### LINKED ISSUE #8190: Resin floor displays weirdly
## Round ID & Round Date:
Round ID : 41724
Round Date : 12.09.2022

## Testmerges:

[#8082: 'Updates our layers to get ready for plane cube, and weather effects' by dwasint at commit b43c4b917d](https://www.github.com/beestation/beestation-hornet/pull/8082)
[#8189: 'Removes some major sleepers, Refactors maploading to be async (100% lag free) 4' by PowerfulBacon at commit 884c35fcbd](https://www.github.com/beestation/beestation-hornet/pull/8189)
[#8123: '[Port-ish] Refactor icon smoothing to use 8-bit bitmasking + new tables and reworked glass icons!' by PestoVerde322 at commit 5740aa7afa](https://www.github.com/beestation/beestation-hornet/pull/8123)

## Reproduction:

Get Xeno turf one way or another, see this : 

image

### LINKED ISSUE #8212: Alien nest is invisible
Issue reported from  Round ID: 41780 (BeeStation Sage - Roleplay)

Reporting client version: 514.1585

Just as the title says.
image

### PR DESCRIPTION
**Pull request by PestoVerde322**
## About The Pull Request

This PR aim to fix some oversights that I've left from the 8bit smoothing PR really. Mainly the Alien weed looking weird and odd smoothing related to some shuttle walls, as well some missing shuttle engines
 
fixes #8212
fixes #8190
## Why It's Good For The Game

Fixes good

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

immagine

</details>

## Changelog
:cl:
tweak: Adjusted the smoothing of some walls, catwalks and lattices
fix: Fixed the missing icons for the resin nest
fix: fixed (hopefully) the broken alien resin bug
fix: bamboo floor no longer smooth with glass, somehow
/:cl:

### CODE REVIEW

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Mineral walls should smooth with normal walls. The issue was that shuttles used to not smooth with things around them, even if they were in the same smoothing group. (Walls wouldnt smooth with adjacent walls if they weren't part of the same shuttle)

File: code/game/objects/structures/aliens.dm [hunk 1]
```diff
@@ -121,7 +121,7 @@
 	smoothing_flags = SMOOTH_BITMASK
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WEEDS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WEEDS)
-	transform = matrix(1, 0, -4, 0, 1, -4)
+	//transform = matrix(1, 0, -4, 0, 1, -4)
```
Comment (itsmeow) [inline code/game/objects/structures/aliens.dm]: Why comment and not remove

**Comment by DrDuckedGoose** [comment]
image
Is this addressed too?

**Comment by itsmeow** [comment]
I also noticed a bug: Catwalks try to smooth into lattices, as in they consider lattices part of the catwalk structure and remove edges near it incorrectly

**Comment by dwasint** [comment]
catwalks are in the wrong group currently they are group lattice when they should be group catwalk would fix that issue

**Comment by imsynthex** [comment]
link this to
- https://github.com/BeeStation/BeeStation-Hornet/issues/8190

and

- https://github.com/BeeStation/BeeStation-Hornet/issues/8212

**Comment by itsmeow** [comment]
Windows smooth with bamboo floors according to some

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
I think the mineral walls should stay on the SMOOTH_GROUP_WALLS smoothing_group (they all share closed turf smoothing group anyway)
The problem with it connecting wasnt the fact that the smoothing group was wrong, its that previously shuttles didn't smooth with other shuttles even if they were the same turf. (Standard walls wouldnt smooth with standard walls attached to other shuttles)

**Comment by PowerfulBacon** [comment]
Can you also potentially look at cult walls showing up as cult walls on meson goggles

**Review by PowerfulBacon — DISMISSED** [review]
This will break clockie walls looking like normal walls to mesons

File: code/__DEFINES/icon_smoothing.dm [hunk 2]
```diff
@@ -57,8 +57,8 @@ DEFINE_BITFIELD(smoothing_flags, list(
 #define SMOOTH_GROUP_CARPET_ROYAL_BLACK S_TURF(18)		///turf/open/floor/carpet/royalblack
 #define SMOOTH_GROUP_CARPET_ROYAL_BLUE S_TURF(19)		///turf/open/floor/carpet/royalblue
 #define SMOOTH_GROUP_CARPET_GRIMY S_TURF(20)			///turf/open/floor/carpet/grimy
-#define SMOOTH_GROUP_BAMBOO_FLOOR S_TURF(52) 			//![/turf/open/floor/bamboo]
-#define SMOOTH_GROUP_CATWALK  S_OBJ(31) 				///obj/structure/lattice/catwalk
+#define SMOOTH_GROUP_BAMBOO_FLOOR S_TURF(21) 			//![/turf/open/floor/bamboo]
+#define SMOOTH_GROUP_CATWALK  S_OBJ(22) 				///obj/structure/lattice/catwalk
```
Comment (Penwin0) [inline code/__DEFINES/icon_smoothing.dm:61]: 21,22....24?

Comment (PestoVerde322) [reply code/__DEFINES/icon_smoothing.dm:61]: Quoting from Marvin  (Wanders, 0.1, MACro) over from discord:

> I think I figured it out. the Smooth Group for bamboo is number 52, while the "maximum" for turf is SMOOTH_GROUP_SURVIVAL_TITANIUM_WALLS  at a cool number 30. Glass is OBJ(21), so 30 + 1 + 21 is 52, the same group as bamboo floors.
The 52 should just probably be 21, and the catwalk below it should probably be 22

**Review by itsmeow — APPROVED** [review]
I am very tired of this being broken, so I am just going to merge it

**Pull request merged**